### ISSUE TITLE
Sometimes Phalcon\Paginator\Adapter\Model next property provides unexpected value

### ISSUE THREAD

**Issue by baryshev** [phalcon/cphalcon#2510]
OS: Mac OS 10.9.3
PHP: 5.5.13
Phalcon: 1.3.2 (Stable)

For several combinations of `Paginator` parameters `next` property provides unexpected value.

For example:

```php
$paginator = new Model([
    'data' => Posts::find(),
    'limit' => 2,
    'page' => 3
]);
```

```
current = 3
before = 2
next = 3 // 4 expected!
last = 4
total_pages = 4
total_items = 7
```

I see this behaviour on penultimate and last pages. `next` property have (`last` - 1) value instead of `last`.

But for some other parameters everything is ok. For example, if i change limit from `2` to `5` everything works as expected.

**Comment by Green-Cat** [comment]
Working on it

**Comment by Green-Cat** [comment]
Fixed in #10114.
Could you try again after the pull request is merged?

**Comment by Green-Cat** [comment]
@phalcon fixed in #10114

**Comment by Green-Cat** [comment]
@andresgutierrez fixed in #10114
